### PR TITLE
Power management

### PR DESCRIPTION
**Pull request by amackworth**
Initial version! Doesn't support dynamic priorities yet, but it works very well in the emulator. (Testing on robot ASAP...)

### CODE REVIEW

File: ZukoAzula2016/src/org/team1540/zukoazula/PIDTalon.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public PIDTalon(TalonExtendedMotor tem, String name, FloatInput targetSpeed) {
         Cluck.publish(name + " Position", tem.modEncoder().getEncoderPosition());
     }
 
-    private static BooleanInput getThreeState(BooleanInput forceTrue, BooleanInput forceFalse) {
+    public static BooleanInput getThreeState(BooleanInput forceTrue, BooleanInput forceFalse) {
```
Comment (AidanSmith) [inline ZukoAzula2016/src/org/team1540/zukoazula/PIDTalon.java:55]: I find it kind of weird to keep this in PIDTalon. Maybe move it to ZukoAzula, or somewhere else? I'm not sure.

Comment (amackworth) [inline ZukoAzula2016/src/org/team1540/zukoazula/PIDTalon.java:55]: We're gonna extract it to the CCRE ASAP.

Comment (AidanSmith) [inline ZukoAzula2016/src/org/team1540/zukoazula/PIDTalon.java:55]: :ok_hand:

File: ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import ccre.timers.StopwatchTimer;
+
+public class PowerManager {
+    private static Map<Integer, List<FloatCell>> filters = new HashMap<Integer, List<FloatCell>>();
+    private static StopwatchTimer timer = new StopwatchTimer();
+
+    public static FloatCell spikePeak = new FloatCell(50);
+    public static FloatCell spikeLength = new FloatCell(2);
+    public static FloatCell degree = new FloatCell((float) 1.3);
+    public static FloatCell scalingFactor = new FloatCell(8);
```
Comment (AidanSmith) [inline ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java]: Shouldn't these be tunable?

File: ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    public static FloatOutput managePower(int priority, FloatOutput motor) {
```
Comment (AidanSmith) [inline ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java:33]: When are you going to add this to the other classes?

Comment (amackworth) [inline ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java:33]: Later. :stuck_out_tongue:

File: ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java (hunk 2 above)
Comment (amackworth) [inline ZukoAzula2016/src/org/team1540/zukoazula/PowerManager.java]: Oh, yep.

**Comment by AidanSmith** [comment]
:+1:

**Pull request merged**